lib/benchmark/bench.go

```diff
@@ -52,62 +57,100 @@ type BenchmarkResult struct {
 	Histogram *hdrhistogram.Histogram
 	// LastError contains last recorded error
 	LastError error
+	// Duration it takes for the whole benchmark to run
+	Duration time.Duration
+}
+
+// ProgressiveBenchmark runs progressive load benchmarking
+func (c *Config) ProgressiveBenchmark(ctx context.Context, tc *client.TeleportClient) (*Result, error) {
```

Comment:
I don't see how this is progressive load..  It appears to only run a single benchmark.  I think the logic here can probably be unified with `Benchmark` -- see my earlier remark about unifying `Duration` and `MinimumWindow`.